File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章统计数字.py
#统计数字 lanqiaoOJ 535

n = int(input())
nums = {}
for i in range(n):
    x = int(input())
    if x in nums.keys():   nums[x] += 1
    else:                  nums[x] = 1
key = list(nums.keys())
key.sort()
for i in key: print(i, nums[i])

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章统计数字.cpp
//统计数字 lanqiaoOJ 535 
#include<bits/stdc++.h>
using namespace std;
int nums[200010];
int main() {
    int n;    scanf("%d",&n);
    for(int i = 1; i <= n; i++)    scanf("%d",&nums[i]);
    sort(nums+1, nums+1+n);
    int cnt = 0;
    for(int i = 1; i <= n; i++) {
        cnt++;
        if(nums[i] != nums[i+1]) {
           printf("%d %d\n", nums[i], cnt);
           cnt = 0;
        }
    }
}
